pkg/services/scm_provider/github.go

```diff
@@ -102,10 +115,10 @@ func (g *GithubProvider) RepoHasPath(ctx context.Context, repo *Repository, path
 	return true, nil
 }
 
-func (g *GithubProvider) listBranches(ctx context.Context, repo *github.Repository) ([]github.Branch, error) {
+func (g *GithubProvider) listBranches(ctx context.Context, repo *Repository) ([]github.Branch, error) {
 	// If we don't specifically want to query for all branches, just use the default branch and call it a day.
 	if !g.allBranches {
-		defaultBranch, _, err := g.client.Repositories.GetBranch(ctx, repo.Owner.GetLogin(), repo.GetName(), repo.GetDefaultBranch())
+		defaultBranch, _, err := g.client.Repositories.GetBranch(ctx, repo.Organization, repo.Repository, repo.Branch)
```

Comment:
Not all repos are owned by an organization, not sure why you changed these.

Reply:
Hi @coderanger, this function now handles a "scm_provider.Repository".  This is because the listing of branches now occurs after ListRepos, and after the repo level filters have been applied.  At this point inside the listBranches function, the githubRepository was found in ListRepos already, and that repo is transformed into an "scm_provider.Repository" as a return of that function call (ListRepos).  Since branches are obtained after ListRepos now, we are dealing with an "scm_provider.Repository" instead of a github.Repository type.

Reply:
D'oh, hoisted by my own petard then :) (I thought about not naming it Organization in the internal struct originally but figured it would cause more confusion).